processor/groupbytraceprocessor/processor.go

```diff
@@ -154,7 +154,7 @@ func (sp *groupByTraceProcessor) processBatch(batch *singleTraceBatch) error {
 
 	// place the trace ID in the buffer, and check if an item had to be evicted
 	evicted := sp.ringBuffer.put(traceID)
-	if evicted.Bytes() != nil {
+	if evicted.IsValid() {
```

Comment:
I think `Bytes() != nil` is far more readable than `IsValid()`, at least for this case here. Perhaps `IsNil()` would be a better name, if all that `IsValid()` does is checking whether the byte array is nil ?

Reply:
a byte array can never be nil, it can be empty (all zeros). so the check for "!= nil" does not make any sense.

Reply:
`IsEmpty()` then?

Reply:
That is something that needs to get to core, but will consider that name. Can you file an issue?

Reply:
Sure: https://github.com/open-telemetry/opentelemetry-collector/issues/2014

Reply:
This should not block the merge, as pointed in the issue `traceID.IsValid()` is equivalent with `!traceID.IsEmpty()`